lib/engine/game/g_18_zoo.rb

```diff
@@ -0,0 +1,631 @@
+      end
+
+      def or_set_finished
+        update_zootickets_value(@turn, 0)
+      end
+
+      def unowned_purchasable_companies(_company)
+        @available_companies
+      end
+
+      # Game will end at the end of the ORs in the third turn
+      def custom_end_game_reached?
+        @turn == 3
+      end
+
+      def check_distance(route, visits)
+        super
+
+        distance = route.train.distance
+
+        return if distance.is_a?(Numeric)
+
+        cities_visited = visits.count { |v| v.city? || (v.offboard? && v.revenue[:yellow].positive?) }
+
+        return unless cities_visited < 2
+
+        raise GameError, 'Water and external gray doesn\'t count against city limit.'
+      end
+
+      def tile_lays(entity)
+        return super unless @floated_corporation
+
+        current_step = ability_blocking_step
+
+        repetition = if current_step&.is_a?(Engine::Step::G18ZOO::HomeTrackAfterPar)
+                       1
+                     else
+                       case @floated_corporation.par_price.price
+                       when 9
+                         2
+                       when 12
+                         4
+                       else
+                         1
+                       end
+                     end
+        repetition.times.map { |_| { lay: true, upgrade: true } }
+      end
+
+      def event_new_train!
+        @new_train_brought = true
+      end
+
+      def round_description(name, round_number = nil)
+        round_number ||= @round.round_num
+        day = case @turn
+              when 1
+                'Monday'
+              when 2
+                'Tuesday'
+              when 3
+                'Wednesday'
+              end
+
+        case name
+        when 'Draft'
+          name
+        when 'Stock'
+          "#{day} Stock"
+        when 'Operating'
+          "#{day} Operating Round (#{round_number} of #{@operating_rounds})"
+        end
+      end
+
+      def reorder_players
+        return if @round.is_a?(Engine::Round::Draft)
+
+        current_order = @players.dup
+        @players.sort_by! { |p| [-p.cash, current_order.index(p)] }
+        @log << "Priority order: #{@players.reject(&:bankrupt).map(&:name).join(', ')}"
+      end
+
+      def game_end_check
+        triggers = {
+          stock_market: @stock_market.max_reached?,
+          custom: custom_end_game_reached?,
+        }.select { |_, t| t }
+
+        %i[immediate current_round current_or full_or one_more_full_or_set].each do |after|
+          triggers.keys.each do |reason|
+            if game_end_check_values[reason] == after
+              (@turn == (@final_turn ||= @turn + 1)) if after == :one_more_full_or_set
+              return [reason, after]
+            end
+          end
+        end
+
+        nil
+      end
+
+      def end_game!
+        return if @finished
+
+        update_zootickets_value(4, 0)
+
+        super
+      end
+    end
+
+    module G18ZOOMaps
```

Comment:
i don't think this should be done this way just have the maps and config as ruby constant hashes